client/src/components/WebAuthn/Verify.js

```diff
@@ -2,87 +2,198 @@
 
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { base64ToByteArray, byteArrayToBase64 } from 'lib/convert';
 import publicKeyType from 'types/publicKey';
+import { performVerification } from '../../lib/auth';
+
+import CircleTick from '../Icons/CircleTick';
+import CircleWarning from '../Icons/CircleWarning';
+import LoadingIndicator from '../LoadingIndicator';
+
+const fallbacks = require('../../../lang/src/en.json');
 
 class Verify extends Component {
   constructor(props) {
     super(props);
 
-    this.handleStartAuth = this.handleStartAuth.bind(this);
-  }
-
-  initAuth() {
-    const { publicKey, onCompleteVerification } = this.props;
-
-    const parsed = {
-      ...publicKey,
-      challenge: base64ToByteArray(publicKey.challenge),
-      allowCredentials: publicKey.allowCredentials.map(data => ({
-        ...data,
-        id: base64ToByteArray(data.id),
-      })),
+    this.state = {
+      failure: false,
     };
 
-    navigator.credentials.get({ publicKey: parsed })
-      .then(response => {
-        onCompleteVerification({
-          credentials: btoa(JSON.stringify({
-            id: response.id,
-            type: response.type,
-            rawId: byteArrayToBase64(response.rawId),
-            response: {
-              clientDataJSON: byteArrayToBase64(response.response.clientDataJSON),
-              authenticatorData: byteArrayToBase64(response.response.authenticatorData),
-              signature: byteArrayToBase64(response.response.signature),
-              userHandle: response.response.userHandle
-                ? byteArrayToBase64(response.response.userHandle)
-                : null,
-            },
-          })),
-        });
-      })
-      .catch(error => {
-        console.log('nay', error, error.message);
-      });
+    this.startAuth = this.startAuth.bind(this);
+    this.handleRetry = this.handleRetry.bind(this);
   }
 
-  handleStartAuth(event) {
+  handleRetry(event) {
     event.preventDefault();
-    this.initAuth();
+    this.setState({
+      failure: false,
+    });
+  }
+
+  startAuth() {
```

Comment:
I think this could use a doc block to explain what it does (specifically what `performVerification` does)